src/Symfony/Component/Serializer/Deserialize/Splitter/SplitterInterface.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Serializer\Deserialize\Splitter;
+
+/**
+ * Splits a list or a dictionnary in a subset of a given $resource stream and yields tokens.
+ *
+ * @author Mathias Arlaud <mathias.arlaud@gmail.com>
+ *
+ * @experimental in 7.0
```

Comment:
Maybe an internal interface? is there any gain in userland to implement this?

Reply:
This extension point can be compared to the `DecoderInterface`, but for the lazy deserialization.
Therefore, if this is turned to internal, it'll prevent users to add lazy deserialization support of a custom format.
But indeed, I agree that the implementation is quite hard and won't be done by regular developers, but maybe it'll at some point needed by bundles?

Reply:
We should not guess which will be the percentage of devs that will do something: Symfony is great because when you want to do something, then you’ll can do it.

If we start to guess, we will close future doors.

we should not guess until it is the only last resort for something really important to have, not for the sake of guessing or doing something that seems to be formally correct.

Also, the other point of view, is that this approach may make harder to respect BC.

so, in the end, we should avoid to guess until it is absolutely necessary.

Reply:
Let's not introduce complexity until proved needed. Adding new extension points when requested by users is easy. Removing them is almost impossible and maintaining them is hard.